i18n4k-core/build.gradle.kts

```diff
@@ -157,54 +109,77 @@ kotlin {
             dependsOn(nativeCommonTest)
         }
 
-        val watchosArm32Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val watchosArm32Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-        val watchosArm64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val watchosArm64Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-        val watchosX86Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val watchosX86Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-          val watchosX64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val watchosX64Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-        val watchosSimulatorArm64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val watchosSimulatorArm64Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-
-        val tvosArm64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val tvosArm64Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-        val tvosX64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val tvosX64Test by getting {
-            dependsOn(nativeCommonTest)
-        }
-        val tvosSimulatorArm64Main by getting {
-            dependsOn(nativeCommonMain)
-        }
-        val tvosSimulatorArm64Test by getting {
-            dependsOn(nativeCommonTest)
+        // Apple native sources
+
+        /* iOS */
+        val iosArm32Main by getting
+        val iosArm32Test by getting
+        val iosArm64Main by getting
+        val iosArm64Test by getting
+        val iosX64Main by getting
+        val iosX64Test by getting
+        val iosSimulatorArm64Main by getting
+        val iosSimulatorArm64Test by getting
+
+        /* Mac OS */
+        val macosX64Main by getting
+        val macosX64Test by getting
+        val macosArm64Main by getting
+        val macosArm64Test by getting
+
+        /* Watch OS */
+        val watchosArm32Main by getting
+        val watchosArm32Test by getting
+        val watchosArm64Main by getting
+        val watchosArm64Test by getting
+        val watchosX86Main by getting
+        val watchosX86Test by getting
+        val watchosX64Main by getting
+        val watchosX64Test by getting
+        val watchosSimulatorArm64Main by getting
+        val watchosSimulatorArm64Test by getting
+
+        /* TV OS */
+        val tvosArm64Main by getting
+        val tvosArm64Test by getting
+        val tvosX64Main by getting
+        val tvosX64Test by getting
+        val tvosSimulatorArm64Main by getting
+        val tvosSimulatorArm64Test by getting
+
+        val iosMain by getting {
```

Comment:
Maybe create an extra source set "appleCommonMain" / "appleCommonTest" that depends on "nativeCommonMain" / "nativeCommonTest" 

The "iosMain" / "iosTest" (which than depends on "appleCommon") should only contain the different "ios" targets.